internal/run/run.go

```diff
@@ -506,12 +505,41 @@ func (w *Workflow) validateDocument(ctx context.Context, parentStep *WorkflowSte
 	return res, err
 }
 
-func (w *Workflow) snapshotSource(ctx context.Context, parentStep *WorkflowStep, namespaceName string, documentPath string) error {
-	hasSchemaRegistry, _ := auth.HasWorkspaceFeatureFlag(ctx, shared.FeatureFlagsSchemaRegistry)
-	if !hasSchemaRegistry {
+func (w *Workflow) snapshotSource(ctx context.Context, parentStep *WorkflowStep, sourceID string, source workflow.Source, documentPath string) error {
+	if !registry.IsRegistryEnabled(ctx) {
 		return nil
 	}
 
+	namespaceName := sourceID
+	if source.Publish != nil && source.Publish.Location != "" {
```

Comment:
Could we separate this into a method like `getNamespaceName(source)`? If possible it would even be nice to have it on the `souce.Publish` model in `sdk-gen-config`

Reply:
I'll move the parse into sdk-gen-config. It's a little tough to move all of this because sdk-gen-config can't import speakeasy-core (speakeasy-core imports sdk-gen-config) so we couldn't do all the auth checks.